src/_gettsim/transfers/rente.py

```diff
@@ -19,8 +19,104 @@ def sum_ges_rente_priv_rente_m(priv_rente_m: float, ges_rente_m: float) -> float
     return out
 
 
-@policy_info(params_key_for_rounding="ges_rente")
-def ges_rente_vor_grundr_m(
+@policy_info(end_date="2020-12-31")
+def ges_rente_m(ges_rente_vor_grundr_m: float) -> float:
+    return ges_rente_vor_grundr_m
+
+
+@policy_info(
+    start_date="2021-01-01",
+    name_in_dag="ges_rente_m",
+    params_key_for_rounding="ges_rente",
+)
+def ges_rente_m_nach_grundr(
+    ges_rente_vor_grundr_m: float,
+    grundr_zuschlag_m: float,
+    rentner: bool,
+) -> float:
+    """Calculate total individual public pension including Grundrentenzuschlag. Is only
+    active after 2021 when Grundrente is in place.
+
+    Parameters
+    ----------
+    ges_rente_vor_grundr_m
+        See :func:`ges_rente_vor_grundr_m`.
+    grundr_zuschlag_m
+        See :func:`grundr_zuschlag_m`.
+    rentner
+        See basic input variable :ref:`rentner <rentner>`.
+
+    Returns
+    -------
+
+    """
+    # Return 0 if person not yet retired
+    out = ges_rente_vor_grundr_m + grundr_zuschlag_m if rentner else 0.0
+    return out
+
+
+@policy_info(
+    end_date="2022-12-31",
+    name_in_dag="ges_rente_vor_grundr_m",
+    params_key_for_rounding="ges_rente",
+)
+def ges_rente_vor_grundr_mit_hinzuverdienstgrenzen_m(
+    bruttorente_ohne_einkommensanrechnung_m: float,
+    bruttorente_mit_einkommensanrechnung_m: float,
+    alter: int,
+    ges_rente_regelaltersgrenze: float,
+) -> float:
+    """Public pension claim before Grundrentenzuschlag.
+
+    Accounts for earnings test before the normal retirement age (NRA) following §34 SGB
+    VI.
+
+    Parameters
+    ----------
+    bruttorente_ohne_einkommensanrechnung_m
+        See :func:`bruttorente_ohne_einkommensanrechnung_m`.
+    bruttorente_mit_einkommensanrechnung_m
+        See :func:`bruttorente_mit_einkommensanrechnung_m`.
+    alter
+        See :func:`alter`.
+    ges_rente_regelaltersgrenze
+        See :func:`ges_rente_regelaltersgrenze`.
+
+    Returns
+    -------
+
+    """
+    if alter >= ges_rente_regelaltersgrenze:
```

Comment:
Here, I need the age with monthly precision. This hinges on the new interface (you discussed adding an `evaluation_date` argument from which we could calculate the age in #211 and #378).

It would be fine for me to keep it imprecise for now and tackle this later. We could also add another input variable `evaluation_month` or something like this in the meantime. What do you think @hmgaudecker ?

Reply:
Yeah, just leave it for now. Would be an unfortunate point in time to tackle this. Linking #781 so we'll keep it in mind.